### ISSUE TITLE
Node.js Best Practices - How to become a better Node.js developer in 2018

### ISSUE THREAD

**Issue by gergelyke** [gergelyke/gergelyke.github.io#12]

**Comment by ShrawanLakhe** [comment]
For me, one should start writing.  i have just started blogging as well on https://www.nodexplained.com/. I find **Secure Your Node.js Web Application** by **Karl Düüna** extremely informative on security aspects of web application programming. I learnt a lot by reading this book.

**Comment by warnero** [comment]
I've been hunting and hunting to find a good answer to this. Can we use the new ES6 modules stuff? If so, how? If not, why not? I'd love to hear your answer. I just recently saw one article suggest implementing both https://gist.github.com/domenic/4748675 .

**Comment by caraya** [comment]
Look at the date for the gist, the situation has changed since then.

Jake Archibald does a pretty good job explaining [current support for modules in browsers](https://jakearchibald.com/2017/es-modules-in-browsers/) and Contentful does a good job explaining if [we should stop bundling now that modules are a thing](https://www.contentful.com/blog/2017/04/04/es6-modules-support-lands-in-browsers-is-it-time-to-rethink-bundling/)

Those articles and a crapload of testing are a good starting point.

**Comment by warnero** [comment]
Neither of those discusses what I brought up - using ES6 modulse in Node.js (unless I'm missing something). I may not have said so explicitly, but didn't think I had to since the article is titled "How to become a better Node.js developer in 2018".

**Comment by caraya** [comment]
You're right. I misread the original issue and added the wrong links to the discussion. 

These are the links for Node:

- [An Update on ES6 Modules in Node.js](https://medium.com/the-node-js-collection/an-update-on-es6-modules-in-node-js-42c958b890c)
- [ES6 modules, Node.js and the Michael Jackson Solution](https://medium.com/dailyjs/es6-modules-node-js-and-the-michael-jackson-solution-828dc244b8b)
- [ES6 Modules in Node 9.3.0](https://nodejs.org/api/esm.html)
- [Using ES modules natively in Node.js](http://2ality.com/2017/09/native-esm-node.html)

Sorry about the confusion

**Comment by warnero** [comment]
Thanks! I wish some of these had shown up in my searches.
